File: Sorting/10814_나이순 정렬.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool compare(pair<int, string> a, pair<int, string> b) {
	return a.first < b.first;
}

void solution(int n, vector<pair<int, string>> &arr) {
	// 정렬
	stable_sort(arr.begin(), arr.end(), compare);

	// 출력
	for (int i = 0; i < n; i++) 
		cout << arr[i].first << ' ' << arr[i].second << '\n';
	
	return;
}

int main(void) {
	int n;
	vector<pair<int, string>> arr;

	cin >> n;

	for (int i = 0; i < n; i++) {
		int age;
		string name;

		cin >> age >> name;
		arr.push_back({ age,name });
	}

	solution(n, arr);
	return 0;
}

File: Sorting/10814_나이순 정렬.py
#입출력
from sys import stdin
#입력 
input = stdin.readline

n=int(input())

arr=[]
for _ in range(n):
    age,name=input().split()
    arr.append([int(age),name])

#람다식을 이용한 정렬
arr.sort(key=lambda x:x[0])

for a in arr:
    print(*a)